extensions/binding/org.eclipse.smarthome.binding.yeelightblue/src/main/java/org/eclipse/smarthome/binding/yeelightblue/internal/handler/YeelightThingHandler.java

```diff
@@ -0,0 +1,349 @@
+        BluetoothGattCharacteristic characteristic = null;
+        String value = null;
+
+        if (command instanceof HSBType) {
+            HSBType hsb = (HSBType) command;
+            double r = hsb.getRed().doubleValue() * 2.55;
+            double g = hsb.getGreen().doubleValue() * 2.55;
+            double b = hsb.getBlue().doubleValue() * 2.55;
+            double a = hsb.getSaturation().doubleValue();
+            value = String.format("%.0f,%.0f,%.0f,%.0f", r, g, b, a);
+        }
+
+        else if (command instanceof PercentType) {
+            value = ",,," + ((PercentType) command).intValue() + "";
+        }
+
+        else if (command instanceof OnOffType) {
+            value = ",,," + ((OnOffType) command == OnOffType.ON ? 100 : 0) + "";
+        }
+
+        if (value == null) {
+            logger.debug("Unable to convert value!");
+            return;
+        }
+
+        characteristic = gattClient.getCharacteristic(UUID_YEELIGHT_CONTROL);
+        if (characteristic == null) {
+            logger.debug("Unable to find control characteristic!");
+            return;
+        }
+
+        // Terminate the value string with commas - up to 18 characters long
+        for (int cnt = value.length(); cnt < 18; cnt++) {
+            value += ",";
+        }
+        logger.debug("Yeelight conversion: {} to \"{}\"", command, value);
+
+        characteristic.setValue(value);
+        gattClient.writeCharacteristic(characteristic);
+
+        setSleepTimer();
+    }
+
+    // The following timer implements a poll when the device status is changed. It requests the status
+    // update a number of times after the update. This ensures that any delays are accounted for.
+    private Timer timer = null;
+    private TimerTask timerTask = null;
+
+    private class WakeupTimerTask extends TimerTask {
+        private int count = 5;
+
+        @Override
+        public void run() {
+            count--;
+            BluetoothGattCharacteristic readCharacteristic = null;
+            logger.debug("Requesting status ({});", count);
+            readCharacteristic = gattClient.getCharacteristic(UUID_YEELIGHT_STATUS_REQUEST);
+            readCharacteristic.setValue("S");
+            gattClient.writeCharacteristic(readCharacteristic);
+
+            if (count == 0) {
+                timerTask.cancel();
+                timerTask = null;
+            }
+        }
+    }
+
+    public synchronized void setSleepTimer() {
+        // Stop any existing timer
+        resetSleepTimer();
+
+        // Create the timer task
+        timerTask = new WakeupTimerTask();
+
+        // Start the timer
+        timer.scheduleAtFixedRate(timerTask, 100, 200);
+    }
+
+    public synchronized void resetSleepTimer() {
+        // Stop any existing timer
+        if (timerTask != null) {
+            timerTask.cancel();
+        }
+        timerTask = null;
+    }
+
+    public float[] RGBtoHSB(int r, int g, int b, float[] hsbvals) {
```

Comment:
I understand that the purpose of this PR is to apply only the most important changes, but I just want to mention that this method seems to me redundant, as it is already implemented in ESH - see HSBType.fromRGB()